### PR TITLE
Extract DarkenService from UpdateStacksFileService for migration to v…

### PR DESCRIPTION
**Pull request by aaron-collier**
…ersioned objects

Follow up from #911

### CODE REVIEW

File: app/services/darken_service.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  def self.darken(cocina_object)
+    new(cocina_object).darken
+  end
+
+  def initialize(cocina_object)
+    @stacks_druid_path = DruidTools::PurlDruid.new(cocina_object.externalIdentifier, Settings.filesystems.stacks_root).pathname
+  end
+
+  # Remove the object from the stacks filesystem and clean up any empty directories in the druid tree.
+  def darken
```
Comment (justinlittman) [inline app/services/darken_service.rb:15]: Minor suggestion. For a service that only has a single method, name that method `call`.

File: app/services/darken_service.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+# Service for darkening an object.
+# This service removes files from the stacks filesystem in order to make an objct dark.
+class DarkenService
+  def self.darken(cocina_object)
+    new(cocina_object).darken
+  end
+
+  def initialize(cocina_object)
```
Comment (justinlittman) [inline app/services/darken_service.rb:10]: Why pass in the cocina object? Looks like all it needs is the druid.

**Review by justinlittman — APPROVED** [review]
Approve pending consideration of comments.

**Comment by aaron-collier** [comment]
@justinlittman done. Thanks.

**Review by jcoyne — APPROVED** [review]

**Pull request merged**